kafka/perseverant_producer.go

```diff
@@ -0,0 +1,97 @@
+package kafka
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/Financial-Times/go-logger/v2"
+	"github.com/Shopify/sarama"
+)
+
+// perseverantProducer implements the Producer interface.
+// It will attempt to create a new producer instance continuously until successful.
+type perseverantProducer struct {
+	sync.RWMutex
+	brokers  string
+	topic    string
+	config   *sarama.Config
+	producer Producer
+	logger   *logger.UPPLogger
+}
+
+// NewPerseverantProducer creates a new perseverantProducer
+func NewPerseverantProducer(brokers string, topic string, config *sarama.Config, initialDelay time.Duration, retryInterval time.Duration, logger *logger.UPPLogger) (Producer, error) {
```

Comment:
Yes, it would be better.

Reply:
After checking, I moved the perseverant consumer to a config wrapper because the basic consumer expects one. Our producers, though, expect explicit parameters, not a struct. Should we move both producer and perseverant producer to expect config structs?

Reply:
I'm fine with both. It's definitely a style preference so please keep it as you wish.